### PR TITLE
[Docs] contributing-to-e2e: update contributing to e2e docs

### PR DESCRIPTION
**Pull request by jerensl**
**Notes for Reviewers**

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 90df80f5a024e3a7b9dc050f81cf81971cc6017e at: https://66f4191ffcba9700c52ccf38--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit e23779ef8b5191dc9b9900a7fa3f00da9ad7d3b0 at: https://66f557ad9f4c161a37c20b67--meshery-docs-preview.netlify.app

**Comment by dragon-slayer875** [comment]
image
Your?

**Comment by dragon-slayer875** [comment]
image

I think this should be included in the information component to better highlight this.

**Comment by dragon-slayer875** [comment]
image

Capitalize?

**Comment by dragon-slayer875** [comment]
I think this can use a bit more elaboration.
Maybe we can at least tell people what Meshery CLI is?
Just thinking out loud.

**Comment by dragon-slayer875** [comment]
image

And is also?

**Comment by dragon-slayer875** [comment]
image
For playwrights, always try to use a native OS whenever possible. The Docker-based approach is intended only for unsupported OSes and is generally not recommended because it runs on top of Ubuntu images, which can be redundant if you're already using Ubuntu or Windows.

**Review by dragon-slayer875 — COMMENTED** [review]
Left some comments. Let me know if they make sense to you @jerensl .

**Comment by jerensl** [comment]
> image I think this can use a bit more elaboration. Maybe we can at least tell people what Meshery CLI is? Just thinking out loud.

A how to-guide is concerned with work - a task or problem, with a practical goal. Maintain focus on that goal.

Anything else that’s added distracts both you and the user and dilutes the useful power of the guide. Typically, the temptations are to explain or to provide reference for completeness. Neither of these are part of guiding the user in their work. They get in the way of the action; if they’re important, link to them.

A how-to guide serves the work of the already-competent user, whom you can assume to know what they want to do, and to be able to follow your instructions correctly.

Source: https://diataxis.fr/how-to-guides/#how-to-guides

**Comment by dragon-slayer875** [comment]
> are to explain or to provide reference for completeness. Neither of these are part of guiding the user in their work. They get in the way of the action; if the

Makes sense @jerensl . Thanks for the info. :+1:

File: docs/pages/project/contributing/contributing-ui-tests.md [hunk 1]
```diff
@@ -38,33 +41,38 @@ There are several tools to help you to working with environment variables locall
...
 ```
 
+### Meshery CLI
+
+There is also Meshery CLI which can help you run the UI and Server, for more detail, you go to [Meshery CLI documentation](https://docs.meshery.io/project/contributing/contributing-cli-guide#process)
+
 ### Docker Based
 
-Alternatively, a Docker-based setup can be utilized, which simplifies the process, ensures consistency across different environments and also more close to production environment as well compare to native solution. 
+Alternatively, a Docker-based setup can be utilized, simplifying the process, and ensuring consistency across different environments. It is closer to the production environment than the native solution but slower in terms of build time.
```
Comment (ianrwhitney) [inline docs/pages/project/contributing/contributing-ui-tests.md:75]: Hey Jerens, thanks for the changes. Curious as to why native os is flagged as recommended however this description notes this is closer to prod env?

File: ui/tests/e2e/env.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-const REMOTE_PROVIDER_URL = process.env.REMOTE_PROVIDER_URL || 'https://staging-meshery.layer5.io';
+const REMOTE_PROVIDER_URL = process.env.REMOTE_PROVIDER_URL || 'https://meshery.layer5.io';
 
 const USER_EMAIL = process.env.REMOTE_PROVIDER_USER_EMAIL;
 const USER_PASSWORD = process.env.REMOTE_PROVIDER_USER_PASSWORD;
 
 const REMOTE_PROVIDER_USER = {
-  email: USER_EMAIL || 'test-admin@layer5.io',
-  password: USER_PASSWORD || 'test-admin',
+  email: USER_EMAIL,
```
Comment (ianrwhitney) [inline ui/tests/e2e/env.js:8]: Nice! I was curious about these defaults!

File: docs/pages/project/contributing/contributing-ui-tests.md [hunk 3]
```diff
@@ -18,17 +18,20 @@ Before diving into Meshery's testing environment, certain prerequisites are nece
...
+- Kubernetes clusters (Required for connection to kubernetets test cases)
+- Already have [Meshery Adapters](https://docs.meshery.io/concepts/architecture/adapters) up and running (Required for adapters test cases)
 
 ## Setting up environment variable
 
 To run the tests successfully, three environment variables must be configured:  
 • `REMOTE_PROVIDER_USER_EMAIL`: The email associated with your account within your provider.  
 • `REMOTE_PROVIDER_USER_PASSWORD` : The password associated with your account within your provider.  
-• `PROVIDER_TOKEN`: You're provider token, that can be generated from your provider account  
+• `PROVIDER_TOKEN`: You're provider token, can be generated from an account registered within your provider
```
Comment (ianrwhitney) [inline docs/pages/project/contributing/contributing-ui-tests.md:29]: Nit: typo Your vs You’re.

**Review by ianrwhitney — COMMENTED** [review]
Overall lgtm. Thanks

**Comment by l5io** [comment]
🚀 Preview for commit 9a9ccff4d87906cc290e2fb60bb80e07709cbfdc at: https://66faf14e88e7050cc0575dc2--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit c834290608a2e9ac2c24409e7ef3700cac1ed65a at: https://66fb85486f19efab1be27c68--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 59b9b5a89831592035481c142667516f18800410 at: https://66fb8bbf78d611a7a853014e--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 0335510e8cad0cfc6b42d299fb3c50291307ca2b at: https://66fbcdac39bf5206c498d714--meshery-docs-preview.netlify.app

File: ui/tests/e2e/env.js (hunk 2 above)
Comment (jerensl) [reply ui/tests/e2e/env.js:8]: @ianrwhitney is just an old account I guess which is already not valid anymore

**Review by leecalcote — DISMISSED** [review]

File: docs/pages/project/contributing/contributing-ui-tests.md [hunk 4]
```diff
@@ -18,17 +18,20 @@ Before diving into Meshery's testing environment, certain prerequisites are nece
 - A verified account in your choosen provider which integrate with Meshery.
 - A compatible browser such as Chromium, Chrome, or Firefox.
 - Installations of Golang, NodeJS, and Makefiles for Native OS build (Optional for docker based build).
-- Kubernetes clusters (Required for several test cases)
-- Already have [Meshery Adapters](https://docs.meshery.io/concepts/architecture/adapters) up and running (Required for several test cases)
+- Kubernetes clusters (Required for connection to kubernetets test cases)
```
Comment (leecalcote) [inline docs/pages/project/contributing/contributing-ui-tests.md:21]:
```suggestion
- Kubernetes clusters (Required for connection to Kubernetes test cases)
```

**Review by leecalcote — APPROVED** [review]

**Comment by l5io** [comment]
🚀 Preview for commit 70ef971073021e3a1bb114ff52870e7b43de7787 at: https://66fd738290696c3161a369a5--meshery-docs-preview.netlify.app

**Review by leecalcote — APPROVED** [review]
Merging this so that we can have the provider value updated.

**Pull request merged**